staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -232,3 +232,14 @@ func HasObjectMetaSystemFieldValues(meta Object) bool {
 	return !meta.GetCreationTimestamp().Time.IsZero() ||
 		len(meta.GetUID()) != 0
 }
+
+// ResetObjectMetaForStatus forces the meta fields for a status update to match the meta fields
+// for a pre-existing object.  Status updates are for updating status, not annotations
+func ResetObjectMetaForStatus(meta, existingMeta Object) {
+	meta.SetGeneration(existingMeta.GetGeneration())
+	meta.SetSelfLink(existingMeta.GetSelfLink())
+	meta.SetLabels(existingMeta.GetLabels())
+	meta.SetAnnotations(existingMeta.GetAnnotations())
```

Comment:
Does it ever make sense to change an annotation and something in status at the same time? I think yes, unfortunately.

Otherwise controllers that want to do this need to send two updates and wouldn't be able to make an atomic change.